xonsh/built_ins.py

```diff
@@ -637,15 +637,41 @@ def run_subproc(cmds, captured=True):
             hist.last_cmd_out = output
     if not prev_is_proxy and hist.last_cmd_rtn > 0 and ENV.get('RAISE_SUBPROC_ERROR'):
         raise CalledProcessError(hist.last_cmd_rtn, aliased_cmd, output=output)
-    if captured:
+    if captured=='stdout':
         return output
+    elif captured=='object':
+        return CompletedProcess(prev_proc, output)
 
 
-def subproc_captured(*cmds):
+class CompletedProcess:
```

Comment:
One could also, I think, make an argument for not relying on the `CompletedProcess` constructor, which isn't documented, and rather using our own, for the sake of clarity.  An alternative would be to copy over the `CompletedProcess` code...or just use a tuple...